src/services/ForecastApiService.js

```diff
@@ -20,19 +20,105 @@ const ApiService = {
     return resultToken;
   },
 
-  getLocationsSearch: async function (url, accessToken) {
+  _getData: async function (url, accessToken) {
     let responseData = {
       status: false
     };
 
     try {
-      const fetchResponse = await ForecastApi.fetchLocationSearch(url, accessToken);
+      const fetchResponse = await ForecastApi.fetchData(url, accessToken);
       responseData = await fetchResponse.json();
       responseData.status = true;
     } catch (error) {
       console.error(error);
     }
 
+    return responseData;
+  },
+
+  getLocationsSearch: async function (locationQueryStr, cookies) {
+    let responseData = {
+      status: false,
+      locations: []
+    };
+
+    try {
+      const accessToken = await ApiService.getAccessToken(cookies);
+      const url = `/api/v1/location/search/${locationQueryStr}`;
+
+      responseData = await ApiService.getLocationsSearch(url, accessToken);
+    } catch (error) {
+      console.error(error);
+    }
+
+    return responseData;
+  },
+
+  getCurrentForecast: async function (locationId, cookies) {
+    let responseData = {
+      status: false
+    };
+
+    try {
+      const accessToken = await this._getAccessToken(cookies);
+      const url = `/api/v1/current/${locationId}`;
+
+      responseData = await this._getData(url, accessToken);
```

Comment:
so what's the point of having _getData? 

```
getLocationInfo: async function (locationId, cookies) {
    let responseData = {
      status: false
    };

    try {
      const accessToken = await this._getAccessToken(cookies);
      const url = `/api/v1/location/${locationId}`;

      responseData = await this._getData(url, accessToken);
    } catch (error) {
      console.error(error);
    }

    return responseData;
  }
```

or 

```
getLocationInfo: async function (locationId, cookies) {
    try {
      const accessToken = await this._getAccessToken(cookies);
      return await ForecastApi.fetchData(`/api/v1/location/${locationId}`, accessToken);
    } catch (error) {
      console.error(error);
    }
    return responseData;
  }
```

Reply:
Done. I have refactored these methods.